src/query/sql/src/planner/optimizer/heuristic/decorrelate/flatten_plan.rs

```diff
@@ -498,6 +504,19 @@ impl SubqueryRewriter {
             flatten_info,
             need_cross_join,
         )?;
+        // Check if sort contains `count() or distinct count()`.
+        if sort.items.iter().any(|item| {
+            let metadata = self.metadata.read();
+            let col = metadata.column(item.index);
+            if let ColumnEntry::DerivedColumn(derived_col) = col {
+                // A little tricky here, we'll check if sort items if a count aggregation function later.
```

Comment:
typo: if sort items if a count aggregation function later. => if sort item is a count aggregation function later.
rest LGTM.